governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -687,4 +694,112 @@ export class PythStakingClient {
+    const targetAccount =
+      await this.stakingProgram.account.targetMetadata.fetch(
+        getTargetAccountAddress(),
+      );
+    return convertBNToBigInt(targetAccount);
+  }
+
+  /**
+   * This returns the current scaling factor between staked tokens and realms voter weight.
+   * The formula is n_staked_tokens = scaling_factor * n_voter_weight
+   */
+  public async getScalingFactor(): Promise<number> {
+    const targetAccount = await this.getTargetAccount();
+    return Number(targetAccount.locked) / MAX_VOTER_WEIGHT;
+  }
+
+  public async getRecoverAccountInstruction(
+    stakeAccountPositions: PublicKey,
+    governanceAuthority: PublicKey,
+  ): Promise<TransactionInstruction> {
+    return this.stakingProgram.methods
+      .recoverAccount()
+      .accountsPartial({
+        stakeAccountPositions,
+        governanceAuthority,
+      })
+      .instruction();
+  }
+
+  public async getUpdatePoolAuthorityInstruction(
+    governanceAuthority: PublicKey,
+    poolAuthority: PublicKey,
+  ): Promise<TransactionInstruction> {
+    return this.stakingProgram.methods
+      .updatePoolAuthority(poolAuthority)
+      .accounts({
+        governanceAuthority,
+      })
+      .instruction();
+  }
+
+  public async getUpdateVoterWeightInstruction(
+    stakeAccountPositions: PublicKey,
+    action: VoterWeightAction,
+    remainingAccount?: PublicKey,
+  ) {
+    return this.stakingProgram.methods
+      .updateVoterWeight(action)
+      .accounts({
+        stakeAccountPositions,
+      })
+      .remainingAccounts(
+        remainingAccount
+          ? [
+              {
+                pubkey: remainingAccount,
+                isWritable: false,
+                isSigner: false,
+              },
+            ]
+          : [],
+      )
+      .instruction();
+  }
+
+  public async getMainStakeAccount() {
+    const stakeAccountPositions = await this.getAllStakeAccountPositions();
+    const stakeAccountBalances = await Promise.all(
+      stakeAccountPositions.map(async (position) => {
+        const stakeAccountCustody = await this.getStakeAccountCustody(position);
+        return {
+          stakeAccountPosition: position,
+          balance: stakeAccountCustody.amount,
+        };
+      }),
+    );
+
+    let mainAccount = stakeAccountBalances[0];
+
+    if (mainAccount === undefined) {
+      return;
+    }
+
+    for (let i = 1; i < stakeAccountBalances.length; i++) {
+      const currentAccount = stakeAccountBalances[i];
+      if (
+        currentAccount !== undefined &&
+        currentAccount.balance > mainAccount.balance
+      ) {
+        mainAccount = currentAccount;
+      }
+    }
+
+    return mainAccount;
+  }
+
+  public async getVoterWeight() {
+    const mainAccount = await this.getMainStakeAccount();
+
+    if (mainAccount === undefined) {
+      return 0;
+    }
+
+    const scalingFactor = await this.getScalingFactor();
+    return Number(mainAccount.balance) / scalingFactor;
```

Comment:
It needs to be the voter weight, that is the sum of all positions that have governance target and are active

Reply:
right!